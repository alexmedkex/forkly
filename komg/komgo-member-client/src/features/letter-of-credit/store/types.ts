import { Action } from 'redux'
import { IDataLetterOfCredit, ILetterOfCredit, LetterOfCreditTaskType } from '@komgo/types'
import { ImmutableMap } from '../../../utils/types'
import { List, Map } from 'immutable'
import { Task } from '../../tasks/store/types'

export enum LetterOfCreditActionType {
  CREATE_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/CREATE_STANDBY_LETTER_OF_CREDIT_REQUEST',
  CREATE_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/CREATE_STANDBY_LETTER_OF_CREDIT_SUCCESS',
  CREATE_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/CREATE_STANDBY_LETTER_OF_CREDIT_FAILURE',
  FETCH_LETTERS_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/FETCH_LETTERS_OF_CREDIT_REQUEST',
  FETCH_LETTERS_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/FETCH_LETTERS_OF_CREDIT_SUCCESS',
  FETCH_LETTERS_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/FETCH_LETTERS_OF_CREDIT_FAILURE',
  GET_LETTER_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/GET_LETTER_OF_CREDIT_REQUEST',
  GET_LETTER_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/GET_LETTER_OF_CREDIT_SUCCESS',
  GET_LETTER_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/GET_LETTER_OF_CREDIT_FAILURE',
  ISSUE_LETTER_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/ISSUE_LETTER_OF_CREDIT_REQUEST',
  ISSUE_LETTER_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/ISSUE_LETTER_OF_CREDIT_SUCCESS',
  ISSUE_LETTER_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/ISSUE_LETTER_OF_CREDIT_FAILURE',
  REJECT_LETTER_OF_CREDIT_REQUEST = '@@templated-letter-of-credit/REJECT_LETTER_OF_CREDIT_REQUEST',
  REJECT_LETTER_OF_CREDIT_SUCCESS = '@@templated-letter-of-credit/REJECT_LETTER_OF_CREDIT_SUCCESS',
  REJECT_LETTER_OF_CREDIT_FAILURE = '@@templated-letter-of-credit/REJECT_LETTER_OF_CREDIT_FAILURE'
}

export type ILetterOfCreditWithData = ILetterOfCredit<IDataLetterOfCredit>

export interface FetchLettersOfCreditSuccessAction extends Action {
  type: LetterOfCreditActionType.FETCH_LETTERS_OF_CREDIT_SUCCESS
  payload: { limit: number; skip: number; total: number; items: ILetterOfCreditWithData[] }
}

export interface GetLetterOfCreditSuccessAction extends Action {
  type: LetterOfCreditActionType.GET_LETTER_OF_CREDIT_SUCCESS
  payload: ILetterOfCreditWithData
}

export interface TemplateStateProperties {
  byStaticId: Map<string, Map<keyof ILetterOfCreditWithData, any>>
  staticIds: List<string>
  total: number
}

export type TemplatedLetterOfCreditState = ImmutableMap<TemplateStateProperties>

export interface ILetterOfCreditEnriched extends ILetterOfCredit<IDataLetterOfCredit> {
  role: string
  actionStatus: string
  latestShipment: string | number | Date
  tasks: Task[]
}

export const LETTER_OF_CREDIT_TYPE_LABELS = {
  [LetterOfCreditTaskType.ReviewIssued]: 'Review issuance',
  [LetterOfCreditTaskType.ReviewRequested]: 'Review request'
}
