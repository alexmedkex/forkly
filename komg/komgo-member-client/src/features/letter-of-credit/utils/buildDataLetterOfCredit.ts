import { IDataLetterOfCredit, ICargo, ITrade, IDataLetterOfCreditBase } from '@komgo/types'
import { Counterparty } from '../../counterparties/store/types'
import { IMember } from '../../members/store/types'
import { findCounterpartyByStatic, findMembersByStatic } from '../../letter-of-credit-legacy/utils/selectors'
import { ImmutableObject } from '../../../utils/types'

interface BuildDataLetterOfCreditProps {
  cargo?: ICargo
  trade: ImmutableObject<ITrade>
  applicant: IMember
  beneficiary: IMember
  issuingBanks: Counterparty[]
  beneficiaryBanks?: IMember[]
  dataLetterOfCreditBase: IDataLetterOfCreditBase
}
export const buildDataLetterOfCredit = ({
  cargo,
  trade,
  applicant,
  beneficiary,
  issuingBanks,
  beneficiaryBanks,
  dataLetterOfCreditBase
}: BuildDataLetterOfCreditProps): Partial<IDataLetterOfCredit> => ({
  cargo,
  trade: trade.toJS(),
  applicant: applicant as any,
  beneficiary: beneficiary as any,
  issuingBank: findCounterpartyByStatic(issuingBanks, dataLetterOfCreditBase.issuingBank.staticId) as any,
  beneficiaryBank: dataLetterOfCreditBase.beneficiaryBank
    ? (findMembersByStatic(beneficiaryBanks, dataLetterOfCreditBase.beneficiaryBank.staticId) as any)
    : undefined,
  amount: dataLetterOfCreditBase.amount,
  issuingBankReference: dataLetterOfCreditBase.issuingBankReference,
  currency: dataLetterOfCreditBase.currency,
  expiryDate: dataLetterOfCreditBase.expiryDate
})
