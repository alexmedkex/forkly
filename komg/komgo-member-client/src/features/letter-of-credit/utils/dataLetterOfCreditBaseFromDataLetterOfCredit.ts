import { IDataLetterOfCredit, IDataLetterOfCreditBase } from '@komgo/types'

export const buildDataLetterOfCreditBase = (input: IDataLetterOfCredit): IDataLetterOfCreditBase => {
  return {
    version: input.version,
    applicant: {
      staticId: input.applicant.staticId
    },
    issuingBank: {
      staticId: input.issuingBank.staticId
    },
    beneficiary: {
      staticId: input.beneficiary.staticId
    },
    beneficiaryBank: input.beneficiaryBank ? { staticId: input.beneficiaryBank.staticId } : undefined,
    trade: {
      source: input.trade.source,
      sourceId: input.trade.sourceId
    },
    cargo: input.cargo
      ? {
          source: input.cargo.source,
          cargoId: input.cargo.cargoId,
          sourceId: input.cargo.sourceId
        }
      : undefined,
    amount: input.amount,
    expiryDate: input.expiryDate,
    currency: input.currency,
    issuingBankReference: input.issuingBankReference || ''
  }
}
