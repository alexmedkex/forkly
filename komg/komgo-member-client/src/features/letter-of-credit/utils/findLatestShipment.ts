import { IDataLetterOfCredit, ILetterOfCredit } from '@komgo/types'

export const findLatestShipment = (letter: ILetterOfCredit<IDataLetterOfCredit>) => {
  const { trade } = letter.templateInstance.data
  return (trade && trade.deliveryPeriod && trade.deliveryPeriod.endDate) || ''
}
