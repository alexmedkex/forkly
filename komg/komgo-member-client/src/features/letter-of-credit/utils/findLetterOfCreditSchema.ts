import {
  TRADE_SCHEMA as TRADE_SCHEMA_V2,
  CARGO_SCHEMA as CARGO_SCHEMA_V2,
  COMPANY_BASE_SCHEMA,
  DATA_LETTER_OF_CREDIT_BASE_SCHEMA,
  DATA_LETTER_OF_CREDIT_TEMPLATE_SCHEMA,
  findSchemaById
} from '@komgo/types'

const KNOWN_SCHEMAS = [
  TRADE_SCHEMA_V2,
  CARGO_SCHEMA_V2,
  COMPANY_BASE_SCHEMA,
  DATA_LETTER_OF_CREDIT_BASE_SCHEMA,
  DATA_LETTER_OF_CREDIT_TEMPLATE_SCHEMA
]

export const findLetterOfCreditSchema = (schemaId: string) => {
  const schema = findSchemaById(KNOWN_SCHEMAS, schemaId)
  if (!schema) {
    throw Error(`No schema found for ${schemaId}`)
  }
  return schema
}
