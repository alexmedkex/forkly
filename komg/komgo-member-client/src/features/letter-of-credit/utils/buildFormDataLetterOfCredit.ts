import { ICargo, ITrade, IDataLetterOfCreditBase } from '@komgo/types'
import { IMember } from '../../members/store/types'
import { ImmutableObject } from '../../../utils/types'

interface BuildDataLetterOfCreditBaseForCreateProps {
  cargo?: ICargo
  trade: ImmutableObject<ITrade>
  applicant: IMember
  beneficiary: IMember
}
export const buildFormDataLetterOfCredit = ({
  cargo,
  trade,
  applicant,
  beneficiary
}: BuildDataLetterOfCreditBaseForCreateProps): IDataLetterOfCreditBase => {
  const startAmount = parseFloat(Number(trade.get('quantity') * trade.get('price')).toFixed())

  return {
    version: 1,
    amount: !isNaN(startAmount) ? startAmount : 0.0,
    currency: trade.get('currency'),
    trade: { sourceId: trade.get('sourceId'), source: trade.get('source') },
    applicant: { staticId: applicant.staticId },
    beneficiary: { staticId: beneficiary.staticId },
    issuingBank: { staticId: '' },
    expiryDate: 'Choose a valid expiry date',
    cargo: cargo && { source: cargo.source, sourceId: cargo.sourceId, cargoId: cargo.cargoId }
  }
}
