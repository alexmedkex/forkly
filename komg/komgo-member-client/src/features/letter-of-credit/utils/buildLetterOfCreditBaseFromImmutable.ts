import { ILetterOfCredit, IDataLetterOfCredit, ILetterOfCreditBase, IDataLetterOfCreditBase } from '@komgo/types'
import { ImmutableObject } from '../../../utils/types'

export const buildLetterOfCreditBaseFromImmutable = (
  letterOfCredit: ImmutableObject<ILetterOfCredit<IDataLetterOfCredit>>,
  data: IDataLetterOfCreditBase,
  templateModel: any
): ILetterOfCreditBase<IDataLetterOfCreditBase> => {
  return {
    type: letterOfCredit.get('type'),
    version: letterOfCredit.get('version'),
    templateInstance: {
      version: letterOfCredit.get('templateInstance').get('version'),
      templateStaticId: letterOfCredit.get('templateInstance').get('templateStaticId'),
      template: templateModel,
      templateSchemaId: letterOfCredit.get('templateInstance').get('templateSchemaId'),
      data,
      dataSchemaId: letterOfCredit.get('templateInstance').get('dataSchemaId'),
      bindings: letterOfCredit
        .get('templateInstance')
        .get('bindings')
        .toJS()
    }
  }
}
