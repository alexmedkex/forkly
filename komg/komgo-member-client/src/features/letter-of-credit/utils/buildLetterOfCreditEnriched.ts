import { IDataLetterOfCredit, ILetterOfCredit } from '@komgo/types'
import { findRole } from './findRole'
import { findTasksByLetterOfCreditStaticId } from './taskUtil'
import { TaskStatus, Task } from '../../tasks/store/types'
import { findLatestShipment } from './findLatestShipment'
import { ILetterOfCreditEnriched } from '../store/types'

export const buildLetterOfCreditEnriched = (
  letter: ILetterOfCredit<IDataLetterOfCredit>,
  tasks: Task[],
  companyStaticId: string
) => {
  const role = findRole(letter, companyStaticId)
  const letterTasks = findTasksByLetterOfCreditStaticId(tasks, letter.staticId)
  const enriched: ILetterOfCreditEnriched = {
    ...letter,
    role,
    actionStatus: letterTasks.length ? TaskStatus.ToDo : TaskStatus.Done,
    latestShipment: findLatestShipment(letter),
    tasks: letterTasks
  }
  return enriched
}
