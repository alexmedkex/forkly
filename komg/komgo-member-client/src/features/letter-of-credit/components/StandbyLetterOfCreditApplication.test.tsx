import * as React from 'react'
import { render, fireEvent, RenderResult, wait } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import {
  StandbyLetterOfCreditApplication,
  StandbyLetterOfCreditApplicationProps
} from './StandbyLetterOfCreditApplication'
import { fakeCounterparty, mockDate } from '../../letter-of-credit-legacy/utils/faker'
import { TradeSource, Currency } from '@komgo/types'

const issuingBanks = [
  fakeCounterparty({ isFinancialInstitution: true, commonName: 'Bank 1' }),
  fakeCounterparty({ isFinancialInstitution: true, commonName: 'Bank 2' }),
  fakeCounterparty({ isFinancialInstitution: true, commonName: 'Bank 3' })
]

const applicantId = 'f0bc781d-c98c-4a06-ac12-3e09a0749d7f'
const beneficiaryId = '2af1d7d7-a3e0-485d-9bcd-53d3d0d22589'
const issuingBankId = ''
const tradeSourceId = 'f9c60190-8546-4445-95df-fb440e7cd82d'
const minValidSlateDocument = {
  nodes: [
    {
      object: 'inline',
      type: 'variable',
      data: {
        path: 'applicant.x500Name.CN'
      }
    },
    {
      object: 'inline',
      type: 'variable',
      data: {
        path: 'issuingBank.x500Name.CN'
      }
    },
    {
      object: 'inline',
      type: 'variable',
      data: {
        path: 'beneficiaryBank.x500Name.CN'
      }
    },
    {
      object: 'inline',
      type: 'variable',
      data: {
        path: 'beneficiary.x500Name.CN'
      }
    },
    {
      object: 'inline',
      type: 'variable',
      data: {
        path: 'amount'
      }
    },
    {
      object: 'inline',
      type: 'variable',
      data: {
        path: 'currency'
      }
    },
    {
      object: 'inline',
      type: 'variable',
      data: {
        path: 'expiryDate'
      }
    },
    {
      object: 'inline',
      type: 'variable',
      data: {
        path: 'issuingBankReference'
      }
    }
  ]
}

describe('StandbyLetterOfCreditApplication', () => {
  let testProps: StandbyLetterOfCreditApplicationProps
  let rendered: RenderResult

  beforeEach(() => {
    mockDate().freeze('Tue Nov 13 2018 00:00:00 GMT+0000 (UTC)')

    testProps = {
      initialValues: {
        version: 1,
        applicant: {
          staticId: applicantId
        },
        beneficiary: {
          staticId: beneficiaryId
        },
        issuingBank: {
          staticId: issuingBankId
        },
        trade: {
          source: TradeSource.Komgo,
          sourceId: tradeSourceId
        },
        cargo: {
          source: TradeSource.Komgo,
          sourceId: tradeSourceId,
          cargoId: 'myCargoId'
        },
        amount: 100,
        expiryDate: '2018-01-01',
        currency: Currency.AED
      },
      beneficiaryBanks: [],
      issuingBanks,
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      templateModel: minValidSlateDocument
    }

    rendered = render(<StandbyLetterOfCreditApplication {...testProps} />)
  })
  afterAll(() => {
    mockDate().restore()
  })
  it('matches snapshot', () => {
    expect(rendered.asFragment()).toMatchSnapshot()
  })
  it('shows errors if a slate document errors is given', () => {
    const first = rendered.asFragment()
    rendered.rerender(<StandbyLetterOfCreditApplication {...testProps} templateModel={{}} />)
    expect(first).toMatchDiffSnapshot(rendered.asFragment())
  })
  it('Allows selection of issuing bank', () => {
    fireEvent.click(rendered.getByTestId('issuingBankId'))
    fireEvent.click(rendered.getAllByText(issuingBanks[0].x500Name.CN)[1])

    expect(testProps.onChange).toHaveBeenCalledWith({
      ...testProps.initialValues,
      issuingBank: {
        staticId: 'cf63c1f8-1165-4c94-a8f8-9252eb4f0016'
      }
    })
  })
  it('Allows selection of opening amount', () => {
    fireEvent.focus(rendered.getByDisplayValue('100.00'))
    userEvent.type(rendered.getByDisplayValue('100'), '1')
    fireEvent.blur(rendered.getByDisplayValue('1'))

    expect(testProps.onChange).toHaveBeenCalledWith({
      ...testProps.initialValues,
      amount: 1
    })
  })
  it('Allows selection of currency', () => {
    userEvent.click(rendered.getByTestId('currency'))
    userEvent.click(rendered.getByText(Currency.USD))

    expect(testProps.onChange).toHaveBeenCalledWith({
      ...testProps.initialValues,
      currency: Currency.USD
    })
  })
  it('Allows selection of expiry date', () => {
    const date = '2030-01-01'

    userEvent.type(rendered.getByDisplayValue('2018-01-01'), date)

    expect(testProps.onChange).toHaveBeenCalledWith({
      ...testProps.initialValues,
      expiryDate: date
    })
  })
  it('calls submit handler if submission button is clicked', async () => {
    fireEvent.click(rendered.getByTestId('issuingBankId'))
    fireEvent.click(rendered.getAllByText(issuingBanks[0].x500Name.CN)[1])

    fireEvent.focus(rendered.getByDisplayValue('100.00'))
    const amount = 1000
    userEvent.type(rendered.getByDisplayValue('100'), `${amount}`)
    fireEvent.blur(rendered.getByDisplayValue('1000'))

    userEvent.click(rendered.getByTestId('currency'))
    userEvent.click(rendered.getByText(Currency.USD))

    const date = '2030-01-01'
    userEvent.type(rendered.getByDisplayValue('2018-01-01'), date)

    fireEvent.click(rendered.getByTestId('submit-letter-of-credit-application'))

    await wait(() => {
      expect(testProps.onSubmit).toHaveBeenCalledWith({
        amount,
        applicant: { staticId: applicantId },
        beneficiary: { staticId: beneficiaryId },
        cargo: {
          cargoId: testProps.initialValues.cargo.cargoId,
          source: testProps.initialValues.cargo.source,
          sourceId: tradeSourceId
        },
        currency: Currency.USD,
        expiryDate: date,
        issuingBank: { staticId: issuingBanks[0].staticId },
        trade: { source: testProps.initialValues.trade.source, sourceId: tradeSourceId },
        version: 1
      })
    })
  })
  it('cant call submit handler if disableSubmit prop is true', async () => {
    rendered.rerender(<StandbyLetterOfCreditApplication {...testProps} disableSubmit={true} />)
    fireEvent.click(rendered.getByTestId('issuingBankId'))
    fireEvent.click(rendered.getAllByText(issuingBanks[0].x500Name.CN)[1])

    fireEvent.focus(rendered.getByDisplayValue('100.00'))
    const amount = 1000
    userEvent.type(rendered.getByDisplayValue('100'), `${amount}`)
    fireEvent.blur(rendered.getByDisplayValue('1000'))

    userEvent.click(rendered.getByTestId('currency'))
    userEvent.click(rendered.getByText(Currency.USD))

    const date = '2030-01-01'
    userEvent.type(rendered.getByDisplayValue('2018-01-01'), date)

    fireEvent.click(rendered.getByTestId('submit-letter-of-credit-application'))

    await wait(() => {
      expect(testProps.onSubmit).not.toHaveBeenCalled()
    })
  })
})
