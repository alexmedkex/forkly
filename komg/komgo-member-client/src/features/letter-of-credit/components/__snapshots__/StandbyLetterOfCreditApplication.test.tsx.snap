// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandbyLetterOfCreditApplication matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  font-size: small;
  color: #d45c64;
  margin-top: -1em;
  padding-top: 3px;
  min-height: 22px;
  margin-bottom: 9px;
}

<form
    class="ui form"
    id="create-standby-letter-of-credit"
    style="display: flex; flex-direction: column; flex-grow: 1;"
  >
    <div
      class="field"
    >
      <label
        class="inputLabel"
      >
        Issuing bank
      </label>
      <div
        aria-expanded="false"
        class="ui compact search selection dropdown dropdown-input"
        data-test-id="issuingBankId"
        name="issuingBank.staticId"
        role="combobox"
        style="width: 100%;"
      >
        <input
          aria-autocomplete="list"
          autocomplete="off"
          class="search"
          id="field_issuingBank.staticId"
          tabindex="1"
          type="text"
          value=""
        />
        <div
          aria-live="polite"
          class="text"
          role="alert"
        />
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
          role="listbox"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Bank 1
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Bank 2
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Bank 3
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c0"
      data-test-id="issuingBank.staticId-error"
    />
    <div
      class="field"
    >
      <label
        class="inputLabel"
      >
        Advising bank
      </label>
      <div
        aria-disabled="true"
        aria-expanded="false"
        class="ui disabled compact selection dropdown field disabled dropdown-input"
        name="beneficiaryBank.staticId"
        role="listbox"
        style="width: 100%;"
        tabindex="-1"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          No advising bank
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              No advising bank
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c0"
      data-test-id="beneficiaryBank.staticId-error"
    />
    <div
      class="field"
    >
      <label>
        Opening amount / Currency
      </label>
      <div
        class="ui right labeled input"
        data-test-id="amount"
      >
        <input
          id="field_amount"
          name="amount"
          tabindex="3"
          type="text"
          value="100.00"
        />
        <div
          aria-expanded="false"
          class="ui dropdown label"
          data-test-id="currency"
          name="currency"
          role="listbox"
          style="border: 1px solid rgb(226, 226, 226);"
          tabindex="0"
        >
          <div
            aria-live="polite"
            class="text"
            role="alert"
          >
            AED
          </div>
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <div
              aria-checked="true"
              aria-selected="true"
              class="active selected item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                AED
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                CHF
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                EUR
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                GBP
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                JPY
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                USD
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c0"
      data-test-id="amount-error"
    />
    <div
      class="field"
    >
      <label>
        Expiry date
      </label>
      <div
        class="field"
        style="display: inherit; justify-content: unset; align-items: center;"
      >
        <label
          class="inputLabel"
        />
        <div
          class="ui input"
          data-test-id="expiryDate"
          style="width: 100%;"
        >
          <input
            id="field_expiryDate"
            min="2018-11-13"
            name="expiryDate"
            tabindex="4"
            type="date"
            value="2018-01-01"
          />
        </div>
      </div>
    </div>
    <div
      class="c0"
      data-test-id="expiryDate-error"
    />
    <button
      class="ui primary disabled button"
      data-test-id="submit-letter-of-credit-application"
      disabled=""
      role="button"
      style="width: 100%; margin-top: auto;"
      tabindex="-1"
      type="submit"
    >
      Submit Application
    </button>
  </form>
</DocumentFragment>
`;

exports[`StandbyLetterOfCreditApplication shows errors if a slate document errors is given 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -12,11 +12,11 @@
        >
          Issuing bank
        </label>
        <div
          aria-expanded=\\"false\\"
-         class=\\"ui compact search selection dropdown dropdown-input\\"
+         class=\\"ui compact search selection dropdown field error dropdown-input\\"
          data-test-id=\\"issuingBankId\\"
          name=\\"issuingBank.staticId\\"
          role=\\"combobox\\"
          style=\\"width: 100%;\\"
        >
@@ -84,12 +84,13 @@
          </div>
        </div>
      </div>
      <div
        class=\\"sc-cIShpX gdPDyy\\"
-       data-test-id=\\"issuingBank.staticId-error\\"
-     />
+     >
+       Field missing in template
+     </div>
      <div
        class=\\"field\\"
      >
        <label
          class=\\"inputLabel\\"
@@ -97,11 +98,11 @@
          Advising bank
        </label>
        <div
          aria-disabled=\\"true\\"
          aria-expanded=\\"false\\"
-         class=\\"ui disabled compact selection dropdown field disabled dropdown-input\\"
+         class=\\"ui disabled compact selection dropdown field error dropdown-input\\"
          name=\\"beneficiaryBank.staticId\\"
          role=\\"listbox\\"
          style=\\"width: 100%;\\"
          tabindex=\\"-1\\"
        >
@@ -135,20 +136,21 @@
          </div>
        </div>
      </div>
      <div
        class=\\"sc-cIShpX gdPDyy\\"
-       data-test-id=\\"beneficiaryBank.staticId-error\\"
-     />
+     >
+       Field missing in template
+     </div>
      <div
        class=\\"field\\"
      >
        <label>
          Opening amount / Currency
        </label>
        <div
-         class=\\"ui right labeled input\\"
+         class=\\"ui right labeled input field error\\"
          data-test-id=\\"amount\\"
        >
          <input
            id=\\"field_amount\\"
            name=\\"amount\\"
@@ -261,12 +263,13 @@
          </div>
        </div>
      </div>
      <div
        class=\\"sc-cIShpX gdPDyy\\"
-       data-test-id=\\"amount-error\\"
-     />
+     >
+       Field missing in template
+     </div>
      <div
        class=\\"field\\"
      >
        <label>
          Expiry date
@@ -277,11 +280,11 @@
        >
          <label
            class=\\"inputLabel\\"
          />
          <div
-           class=\\"ui input\\"
+           class=\\"ui input field error\\"
            data-test-id=\\"expiryDate\\"
            style=\\"width: 100%;\\"
          >
            <input
              id=\\"field_expiryDate\\"
@@ -294,12 +297,13 @@
          </div>
        </div>
      </div>
      <div
        class=\\"sc-cIShpX gdPDyy\\"
-       data-test-id=\\"expiryDate-error\\"
-     />
+     >
+       Field missing in template
+     </div>
      <button
        class=\\"ui primary disabled button\\"
        data-test-id=\\"submit-letter-of-credit-application\\"
        disabled=\\"\\"
        role=\\"button\\""
`;
